### DISCUSSION TITLE
Unicode Encode Error

### DISCUSSION THREAD

**Discussion by alexwillingham98** [thoughtspot/cs_tools#99] [Q&A]
│                                                                                                                    │
│           Info snapshot taken on 2024-01-25                                                                        │
│                                                                                                                    │
│               CS Tools: 1.4.15                                                                                     │
│         Python Version: Python 3.11.7 (tags/v3.11.7:fa7a6f2, Dec  4 2023, 19:24:49) [MSC v.1937 64 bit (AMD64)]    │
│            System Info: Windows (detail: Windows-10-10.0.20348-SP0)                                                │
│      Configs Directory: C:\Users\ {anonymous} \AppData\Roaming\cs_tools                                            │
│    Activate VirtualEnv: C:\Users\ {anonymous} \AppData\Roaming\cs_tools\.cs_tools\Scripts\Activate.ps1             │
│          Platform Tags: win-amd64

We are getting encoding errors when we try executing cs_tools commands in Windows Powershell.  Based on the error, it doesn't like Windows 1252 encoding
(...Python\Lib\encodings\cp1252.py).
 
Error:
File "C:\Users\....\Python\Lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeEncodeError: 'charmap' codec can't encode character '\U0001f4c1' in position 0: character maps to <undefined>
 
Python (3.11.7)
cs_tools (1.4.15)
Previously created Azure VM (Windows Server 2022 Data Center) with a VNET, IP addresses whitelisted, etc.
 
Does anyone have any suggestions to resolve this?

**Comment by boonhapus** [comment]
@alexwillingham98 what command are you typing to see this issue?

**Comment by alexwillingham98** [comment]
s_tools tools searchable bi-server --syncer snowflake:///Users/datafactoryvm/Desktop/cstools/syncer/snowflake.biserver.append.toml --config ali-ts-prod --from-date 2024-01-23 --to-date 2024-01-24

**Reply by alexwillingham98** [reply]
Sorry, should've included that

**Reply by boonhapus** [reply]
Gotcha, thanks for that @alexwillingham98 . Something in that data has either an emoji or a double-byte character (ie Japanese, Arabic, etc). This should be taken care of in CS Tools itself, but the Snowflake Syncer (or rather, underlying Snowflake library itself) may be getting caught on this data.

Are you able to share the log file as well? Re-run the above command and then run `cs_tools logs report` and grab the last log file.

**[https://thoughtspot.egnyte.com/ul/4WLTjVTVUy](https://thoughtspot.egnyte.com/ul/4WLTjVTVUy)**

**Reply by alexwillingham98** [reply]
[2024-01-24T21_51_20.log](https://github.com/thoughtspot/cs_tools/files/14085724/2024-01-24T21_51_20.log)

**Reply by boonhapus** [reply]
So, I think the command you entered was actually `cs_tools config show` -- this is where there error comes from, and it's not liking emojis. What's odd to me is that even a Windows VM should handle this fine. 🤔

@alexwillingham98 I wonder if the VM is running a really old version of Powershell. Can you share the output of `$PSVersionTable.PSVersion` ?

**Reply by alexwillingham98** [reply]
So he updated and then ran the cs_tools command again with the same error, here's the output from the command you sent:
 Major  Minor  Patch  PreReleaseLabel BuildLabel
-----  -----  -----  --------------- ----------
7      4      1

**Reply by alexwillingham98** [reply]
[20240131_log_after_PS7install.log](https://github.com/thoughtspot/cs_tools/files/14116596/20240131_log_after_PS7install.log)
log file after powershell update
